src/Application.php

```diff
@@ -104,7 +104,7 @@ public function pipe($middlewareOrPath, $middleware = null) : void
         $middleware = $middleware ?: $middlewareOrPath;
         $path = $middleware === $middlewareOrPath ? '/' : $middlewareOrPath;
 
-        $middleware = $path !== '/'
+        $middleware = ( is_string($path) && $path !== '/' )
```

Comment:
see #564

Reply:
Sorry wasn't clear. for me. The TypeError looks good. I'll update